pkg/scheduler/scheduler.go

```diff
@@ -608,6 +612,16 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		return
 	}
 
+	// At the end of a successful scheduling cycle, pop and move up Pods if needed.
+	if len(podsToMove.PodsMap) != 0 {
+		var movingPods []*v1.Pod
+		for key, pod := range podsToMove.PodsMap {
+			movingPods = append(movingPods, pod)
+			delete(podsToMove.PodsMap, key)
+		}
+		sched.SchedulingQueue.MoveUpPods(movingPods)
```

Comment:
change the signature to a receive a map to avoid the copy above.

Reply:
done.